client/core/core.go

```diff
@@ -5288,114 +5396,142 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 	c.sentCommits[prefix.Commit] = commitSig
 	c.sentCommitsMtx.Unlock()
 
+	// Prepare order meta data.
+	dbOrder := &db.MetaOrder{
+		MetaData: &db.OrderMetaData{
+			Host:               dc.acct.host,
+			EpochDur:           mktConf.EpochLen, // epochIndex := result.ServerTime / mktConf.EpochLen
+			FromSwapConf:       assetConfigs.fromAsset.SwapConf,
+			ToSwapConf:         assetConfigs.toAsset.SwapConf,
+			MaxFeeRate:         assetConfigs.fromAsset.MaxFeeRate,
+			RedeemMaxFeeRate:   assetConfigs.toAsset.MaxFeeRate,
+			FromVersion:        assetConfigs.fromAsset.Version,
+			ToVersion:          assetConfigs.toAsset.Version, // and we're done with the server's asset configs.
+			Options:            form.Options,
+			RedemptionReserves: redemptionReserves,
+			RefundReserves:     refundReserves,
+			ChangeCoin:         changeID,
+			ProgramID:          form.Program,
+		},
+		Order: ord,
+	}
+
+	tradeRequest := &tradeRequest{
+		mktID:        mktID,
+		route:        route,
+		dc:           dc,
+		form:         form,
+		dbOrder:      dbOrder,
+		msgOrder:     msgOrder,
+		recoveryCoin: recoveryCoin,
+		coins:        coins,
+		wallets:      wallets,
+		errCloser:    errCloser.Copy(),
+		preImg:       preImg,
+	}
+
+	errCloser.Success()
+
+	return tradeRequest, nil
+}
+
+// sendTradeRequest sends an order, processes the result, then prepares and
+// stores the trackedTrade.
+func (c *Core) sendTradeRequest(tr *tradeRequest) (*Order, error) {
+	defer tr.errCloser.Done(c.log)
 	// Send and get the result.
 	result := new(msgjson.OrderResult)
-	err = dc.signAndRequest(msgOrder, route, result, fundingTxWait+DefaultResponseTimeout)
+	err := tr.dc.signAndRequest(tr.msgOrder, tr.route, result, fundingTxWait+DefaultResponseTimeout)
 	if err != nil {
 		// At this point there is a possibility that the server got the request
 		// and created the trade order, but we lost the connection before
 		// receiving the response with the trade's order ID. Any preimage
 		// request will be unrecognized. This order is ABANDONED.
-		return nil, nil, fmt.Errorf("new order request with DEX server %v market %v failed: %w", dc.acct.host, mktID, err)
+		return nil, fmt.Errorf("new order request with DEX server %v market %v failed: %w", tr.dc.acct.host, tr.mktID, err)
 	}
 
 	// If we encounter an error, perform some basic logging.
-	logAbandon := func(err interface{}) {
+	logAbandon := func(err error) {
```

Comment:
This was fine before, but if we were to make a change, we'd make it `msg string` instead, and keep the `fmt.Sprintf` lines.  There's no reason this needs to be a type that implements the `error` interface.  It's just logging.

Although TBH, I don't seen the point in `logAbandon` at all.  We should just log when needing to log (only in two places presently), not bounce around the call stack, worrying about argument types etc.